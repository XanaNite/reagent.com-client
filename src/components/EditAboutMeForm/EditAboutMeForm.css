.editFormSpacing{
    margin: 5px;
}

.editForm-container input[type=text]{
    width: 150px;
    height: 25px;
}

.editForm-container input[type=number]{
    width: 40px;
    height: 25px;
    padding: 2px;
}

.editForm-container span{
    margin-left: 10px;
}

.editForm-container textarea{
    width: 150px;
    height: 100px;
}

.phoneType-container{
    margin: 10px 0px;
    display: flex;
    justify-content: flex-end;
}

.editForm-button{
    text-align: center;
    margin: 25px;
}

.editForm-button button{
    width: 100px;
    border: 0;
    border-radius: 2px;
    height: 40px;
    font-size: 16px;
    margin: 25px;
}