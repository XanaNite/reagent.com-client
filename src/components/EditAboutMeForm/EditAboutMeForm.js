import React from 'react'
import './EditAboutMeForm.css'

export default function EditAboutMeForm(){
    return (
        <div className="editAboutMe-container sectionBorders">
            <form className="editAboutMe-form">
                <h3>Update Your Profile</h3>
                <div className="editAboutMe-container">
                    <div className="newFirstName-container">
                        <label htmlFor="newFirstName">First name</label>
                        <input type="text" name="newFirstName" id="newFirstName" />
                    </div>
                    <div className="newLastName-container">
                        <label htmlFor="newLastName">Last name</label>
                        <input type="text" name="newLastName" id="newLastName" />
                    </div>
                    <div className="newSuffix-container">
                        <label htmlFor="newSuffix">Suffix</label>
                        <input type="text" name="newSuffix" id="newSuffix" />
                    </div>
                    <div className="newNickname-container">
                        <label htmlFor="newNickname">Nickname</label>
                        <input type="text" name="newNickname" id="newNickname" />
                    </div>
                    <div className="newPhone-container">
                        <label htmlFor="newPhone">Phone number</label>
                        <input type="text" name="newPhone" id="newPhone" />
                        <select name="phoneType" id="phoneType">
                            <option value="home">Home</option>
                            <option value="mobile">Mobile</option>
                            <option value="office">Office</option>
                        </select>   
                    </div>
                    <div className="newEmail-container">
                        <label htmlFor="newEmail">Contact email address</label>
                        <input type="text" name="newEmail" id="newEmail" />
                    </div>
                    <div className="newTitle-container">
                        <label htmlFor="newTitle">Professional title</label>
                        <input type="text" name="newTitle" id="newTitle" />
                    </div>
                    <div className="newExperience-container">
                        <label htmlFor="newExperience">Years of experience</label>
                        <input type="text" name="newExperience" id="newExperience" />
                    </div>
                    <div className="newWorkWith-container">
                        <label htmlFor="newWorkWith">Who do you work with?</label>
                        <input type="text" name="newWorkWith" id="newWorkWith" />
                    </div>
                    <div className="newSpecialties-container">
                        <label htmlFor="newSpecialties">Specialties</label>
                        <input type="text" name="newSpecialties" id="newSpecialties" />
                    </div>
                    <div className="newAreas-container">
                        <label htmlFor="newAreas">Areas served</label>
                        <input type="text" name="newAreas" id="newAreas" />
                    </div>
                    <div className="newZip-container">
                        <label htmlFor="newZip">Zip codes served</label>
                        <input type="text" name="newZip" id="newZip" />
                    </div>
                    <div className="newWebsite-container">
                        <label htmlFor="newWebsite">Website</label>
                        <input type="text" name="newWebsite" id="newWebsite" />
                    </div>
                    <div className="newUrl-container">
                        <label htmlFor="newUrl">Video URL</label>
                        <input type="text" name="newUrl" id="newUrl" />
                    </div>
                    <div className="newBio-container">
                        <label htmlFor="newBio">Bio</label>
                        <input type="text" name="newBio" id="newBio" />
                    </div>
                </div>
                <button type="submit">Update</button>
            </form>
        </div>
    )
}