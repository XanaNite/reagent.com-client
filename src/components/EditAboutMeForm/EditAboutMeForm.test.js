import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditAboutMeForm from './EditAboutMeForm';

describe(`EditAboutMeForm component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop',
        agent: {
            first_name: "test",
            last_name: "test",
            title: "test",
            city: "test",
            state: "test"
        }
    };

    it('renders the EditAboutMeForm given props', () => {
        const wrapper = shallow(<EditAboutMeForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})