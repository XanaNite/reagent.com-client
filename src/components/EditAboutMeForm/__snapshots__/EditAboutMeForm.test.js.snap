// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAboutMeForm component renders the EditAboutMeForm given props 1`] = `
<div
  className="editAboutMe-container sectionBorders"
>
  <form
    className="editAboutMe-form"
    onSubmit={[Function]}
  >
    <h3>
      Update Your Profile
    </h3>
    <div
      className="editForm-container"
    >
      <div
        className="editForm__error"
        role="alert"
      />
      <div
        className="newFirstName-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newFirstName"
        >
          First name
        </label>
        <input
          defaultValue="test"
          id="newFirstName"
          name="newFirstName"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newLastName-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newLastName"
        >
          Last name
        </label>
        <input
          defaultValue="test"
          id="newLastName"
          name="newLastName"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newPhone-container editFormSpacing"
      >
        <div
          className="flexbox-between"
        >
          <label
            htmlFor="newPhone"
          >
            Phone number
          </label>
          <input
            id="newPhone"
            name="newPhone"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="phoneType-container"
        >
          <select
            id="phoneType"
            name="phoneType"
            onChange={[Function]}
          >
            <option
              value="home"
            >
              home
            </option>
            <option
              value="mobile"
            >
              mobile
            </option>
            <option
              value="business"
            >
              business
            </option>
            <option
              value="fax"
            >
              fax
            </option>
          </select>
        </div>
      </div>
      <div
        className="newTitle-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newTitle"
        >
          Professional title
        </label>
        <input
          defaultValue="test"
          id="newTitle"
          name="newTitle"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newCity-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newCity"
        >
          City
        </label>
        <input
          defaultValue="test"
          id="newCity"
          name="newCity"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newState-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newState"
        >
          State
        </label>
        <input
          defaultValue="test"
          id="newState"
          name="newState"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newZip-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newZip"
        >
          Zip
        </label>
        <input
          id="newZip"
          name="newZip"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newOffice-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newOffice"
        >
          Office
        </label>
        <input
          id="newOffice"
          name="newOffice"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newExperience-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newExperience"
        >
          Experience
        </label>
        <div>
          <input
            id="newExperience"
            max="75"
            name="newExperience"
            onChange={[Function]}
            type="number"
          />
          <span>
            years
          </span>
        </div>
      </div>
      <div
        className="newBrokerage-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newBrokerage"
        >
          Brokerage
        </label>
        <input
          id="newBrokerage"
          name="newBrokerage"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newSlogan-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newSlogan"
        >
          Slogan
        </label>
        <input
          id="newSlogan"
          name="newSlogan"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="newBio-container flexbox-between editFormSpacing"
      >
        <label
          htmlFor="newBio"
        >
          Bio
        </label>
        <textarea
          id="newBio"
          name="newBio"
          onChange={[Function]}
        />
      </div>
    </div>
    <div
      className="editForm-button"
    >
      <button
        type="submit"
      >
        Update
      </button>
    </div>
  </form>
</div>
`;
